src/synthesiser/Synthesiser.cpp

```diff
@@ -670,13 +678,13 @@ void Synthesiser::emitCode(std::ostream& out, const Statement& stmt) {
         }
 
         void visitScan(const Scan& scan, std::ostream& out) override {
-            const auto& rel = scan.getRelation();
+            const auto& rel = synthesiser.lookup(scan.getRelation());
```

Comment:
```suggestion
            const auto* rel = synthesiser.lookup(scan.getRelation());
```
Similarly for the other refs to pointers

Reply:
Done.